packages/auth-passport/authService.js

```diff
@@ -36,26 +37,37 @@ PassportAuthService.prototype.getProfile = function() {
   var self = this;
   return this.http(req, { withCredentials: true }).then(function(res) {
     userProfile = res.data;
+    localStorage.setItem(USER_CACHE_KEY, JSON.stringify(userProfile));
     return userProfile;
   }).catch(function(err) {
     if (err.status === 401) {
       this.window.location = cloudUrl + CONSTANTS.LOGIN_URL;
-    } else if (err.status === 403) {
+      return;
+    }
+    if (err.status === 403) {
       self.dialog.show(self.dialog.alert({
         title: 'Forbidden',
         textContent: 'You are not authorized to access this resource.',
         ok: 'OK'
       }));
-    } else if (err.status === -1) {
+      return;
+    }
+    var cachedUser;
+    try {
+      cachedUser = JSON.parse(localStorage.getItem(USER_CACHE_KEY));
+    } catch (err) {
+    }
+    if (err.status === -1 && cachedUser) {
```

Comment:
Really nice way to add localStorage support.
Could you update the function's jsdoc to highlight that it has offline support?

Reply:
That change is only temporary. We going to refactor entire implementation here to support offline page for mobile. This code is also executed for the portal so we will need to get this local storage items out.